src/main/java/frc/robot/controllers/ShooterController.java

```diff
@@ -24,69 +31,103 @@
 
-    public ShooterController() {
-        //initialize parameters
-        shooterTalon = new TalonFX(Context.shooterMotorID);
-        shooterTalon.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
-        velocityJRAD = new JRAD(kF, kI, kLoadRatio); //need tuning
+    public ShooterController(int _leftShooterID, int _rightShooterID, boolean _orientation) {
+        //initialize variables
+
+        leftShooterID = _leftShooterID;
+        rightShooterID = _rightShooterID;
+        orientation = _orientation;
+
+        leftShooterTalon = new TalonFX(leftShooterID);
+        leftShooterTalon.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        rightShooterTalon = new TalonFX(rightShooterID);
+        rightShooterTalon.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+
+        velocityJRADD = new JRADD(kP, kT, kF, kI, kLoadRatio, loadRatioConstant, loadRatioRate);
+
         M_SHOOTING_RADIUS = Context.M_FLYWHEEL_RADIUS + Context.M_BALL_DIAMETER/2;
-        startTime = System.currentTimeMillis()/1000;
+        startTime = System.currentTimeMillis();
     }
 
     private void updateParameters() {
         //updates all necessary
-        actualVelocity = flywheelVelocity()/2; //accounts for fact that ball rolls on inside of hood
+
+        actualVelocity = shootingVelocity();
         lastTime = time;
         time = Context.getRelativeTimeSeconds(startTime);
         deltaTime = time - lastTime;
-        setVelocity = velocityJRAD.update(desiredVelocity, actualVelocity, deltaTime);
+        setVelocity = velocityJRADD.update(desiredVelocity, actualVelocity, deltaTime);
     }
 
     private void updateVelocity() {
         //passes input to motor controller
-        setCurrent = AdditionalMath.Clamp(speedConverter(setVelocity), -0.8, 0.8);
-        shooterTalon.set(ControlMode.PercentOutput, -setCurrent); //Sign depends on motor orientation
+
+        setCurrent = AdditionalMath.Clamp(speedConverter(setVelocity), -MAX_CURRENT, MAX_CURRENT);
+        leftShooterTalon.set(ControlMode.PercentOutput, -setCurrent); //Sign depends on motor orientation
+        rightShooterTalon.set(ControlMode.PercentOutput, setCurrent);
+    }
+
+    public void startShooting() {
+        //Used to update objects/parameters relying on constant updates after a period without updates
+
+        lastTime = Context.getRelativeTimeSeconds(startTime);
+        velocityJRADD = new JRADD(velocityJRADD.kP, velocityJRADD.kT, velocityJRADD.kF, velocityJRADD.kI, velocityJRADD.kLoadRatio, loadRatioConstant, loadRatioRate);
     }
 
     public void loop() {
         //execute update methods
+
         updateParameters();
         updateVelocity();
     }
     
     public void loop(double desiredVelocity) {
         //change desiredVelocity, and then execute update methods
-        this.desiredVelocity = desiredVelocity;
+
+        this.desiredVelocity = (orientation) ? desiredVelocity : -desiredVelocity;;
```

Comment:
two semicolons

Reply:
Also orientation is not a very good name for the variable. I only know what it is by reading this code. I would recommend calling it isShooterInverted or something like that.